tests/tests.rs

```diff
@@ -144,32 +144,27 @@ fn run_compile_test(
     let crate_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
     let tests_path = Path::new(&crate_dir).join("tests");
     let mut generated_file = tests_path.join("expectations");
-    if let Some(style) = style {
-        match style {
-            Style::Both => {
-                generated_file.push("both");
-            }
-            Style::Tag => {
-                generated_file.push("tag");
-            }
-            Style::Type => {}
-        }
-    }
 
-    let ext = match language {
-        Language::Cxx => "cpp",
-        Language::C => {
-            if cpp_compat {
-                "compat.c"
-            } else {
-                "c"
-            }
-        }
+    let compat_ext = match (language, cpp_compat) {
+        (Language::C, false) => ".nocompat",
```

Comment:
I think this would be slightly nicer keeping `.compat.` as the special case rather than the other way around.